docs/integrations/github_actions/aws_oidc.md

```diff
@@ -0,0 +1,162 @@
+<img alt="aws_iam_role" src="/images/docs/ci-cd-pipelines/oidc/aws_iam_role.png" />
+</div>
+
+You will need to add one GitHub Secret, `OIDC_AWS_ROLE_TO_ASSUME`, which is the ARN of the above IAM role that handles OIDC federation. You can get the IAM Role ARN from the Terraform output or the AWS IAM Console.
+
+<div style={{"marginBottom":"2em","borderWidth":"thin", "borderStyle":"solid", "borderColor":"lightgray", "padding":"20px", "width":"90%"}}>
+<img alt="gh_secret" src="/images/docs/ci-cd-pipelines/oidc/gh_secret.png" />
+</div>
+
+## Define the workflow
+
+Now create a GitHub Actions workflow in your repository, as `.github/workflows/steampipe.yml`, based on [this example](https://github.com/turbot/steampipe-samples/blob/main/all/github-actions-oidc/aws/steampipe-sample-aws-workflow.yml). Let's review the key elements of the workflow.
+
+### On
+
+GitHub supports a variety of driven-triggers. Here we define two: `workflow_dispatch` to [manually run a workflow](https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow) and `schedule` to run on a cron-like schedule.
+
+```yaml
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 4 7,14,21,28 * *"
+```
+
+### Permissions
+
+Every time your job runs, GitHub's OIDC Provider auto-generates an OIDC token. This token contains multiple claims to establish a security-hardened and verifiable identity about the specific workflow that is trying to authenticate. In order to request this OIDC JWT ID token, your job or workflow run requires a permissions setting with `id-token: write`.
+
+In order to checkout to the GitHub repository and to save the benchmark results to the repository, your job or workflow run requires a permissions setting with `contents: write`.
+
+```yaml
+permissions:
+  id-token: write
+  contents: write
+```
+
+### Steps
+
+A workflow comprises one or more jobs that run in parallel, each with one or more steps that run in order. Our [example](https://github.com/turbot/steampipe-samples/blob/main/all/github-actions-oidc/aws/steampipe-sample-aws-workflow.yml#L14) defines a single job with a series of steps that authenticate to AWS, install Steampipe, run a compliance benchmark and save the results to the repository.
+
+Here's the step that configures the credentials Steampipe will use to access AWS.
+
+```yaml
+- name: "Configure AWS credentials"
+  id: config-aws-auth
+  uses: aws-actions/configure-aws-credentials@v1-node16
+  with:
+    role-to-assume: ${{ secrets.OIDC_AWS_ROLE_TO_ASSUME }}
+    role-session-name: "steampipe-demo"
+    role-duration-seconds: 900
+    aws-region: "us-east-1"
+```
+
+Once the cloud provider successfully validates the claims presented in the OIDC JWT ID token, it then provides a short-lived access token that is available only for the duration of the job. The short-lived access token is exported as environment variables like AWS_DEFAULT_REGION, AWS_REGION, AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY and AWS_SESSION_TOKEN.
```

Comment:
I guess depending on the auth method it would id+secret or token, maybe spell out which case is which?

How does the token map to region vars?

Reply:
It sets the AWS_DEFAULT_REGION to us-east-1. Otherwise we can specify the region in the workflow.
Example: https://github.com/turbot/steampipe-samples/blob/main/all/github-actions-oidc/aws/steampipe-sample-aws-workflow.yml#L31